qtaf-core/src/test/java/de/qytera/qtaf/core/log/model/message/AssertionLogMessageTest.java

```diff
@@ -0,0 +1,111 @@
+package de.qytera.qtaf.core.log.model.message;
+
+import de.qytera.qtaf.core.log.model.LogLevel;
+import de.qytera.qtaf.core.log.model.error.ThrowableWrapper;
+import org.openqa.selenium.InvalidArgumentException;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class AssertionLogMessageTest {
+    @Test(description = "Test construction of AssertionLogMessage")
+    public void testAssertionLogMessageConstructor() {
+        AssertionLogMessage message = new AssertionLogMessage(LogLevel.INFO, "m1");
+        Assert.assertEquals(message.getLevel(), LogLevel.INFO);
+    }
+
+    @Test(description = "Test log levels of AssertionLogMessage")
+    public void testLogLevels() {
+        AssertionLogMessage message = new AssertionLogMessage(LogLevel.INFO, "m1");
+        Assert.assertEquals(message.getLevel(), LogLevel.INFO);
+        message.setLevel(LogLevel.DEBUG);
+        Assert.assertEquals(message.getLevel(), LogLevel.DEBUG);
+        message.setLevel(LogLevel.WARN);
+        Assert.assertEquals(message.getLevel(), LogLevel.WARN);
+        message.setLevel(LogLevel.ERROR);
+        Assert.assertEquals(message.getLevel(), LogLevel.ERROR);
+    }
+
+    @Test(description = "Test status of AssertionLogMessage")
+    public void testStatus() {
+        AssertionLogMessage message = new AssertionLogMessage(LogLevel.INFO, "m1");
+        Assert.assertFalse(message.hasPassed());
+        Assert.assertFalse(message.hasFailed());
```

Comment:
I feel like this could be confusing. Intuitively, I would expect `hasFailed()` to be `true` if `hasPassed()` is `false`. It was surprising for me that `null` is possible as well (internally). Do you think it might make sense to throw an `IllegalStateException` in both methods if no status was ever set?

Reply:
I added an additioal method named 'wasExecuted' that checks if the Assertion was already executed, that means if the status is not null.